### PR TITLE
[aggregator] Remove aggregator API gating

### PR DESCRIPTION
**Pull request by georgemitenkov**
## Description

This PR removes the usage of aggregator API feature flag (it can be therefore re-used in the future) as #13247 changes the Move code to not rely on it either. API has been fully rolled out as well.

The changes are tested by replay (latest):
https://github.com/aptos-labs/aptos-core/actions/runs/9300355540 with few jobs failing because of timeouts.

## Type of Change
- [ ] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Documentation update
- [ ] Tests

## Which Components or Systems Does This Change Impact?
- [ ] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [ ] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Other (specify)

## Checklist
- [ ] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I identified and added all stakeholders and component owners affected by this change as reviewers
- [ ] I tested both happy and unhappy path of the functionality
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by georgemitenkov** [comment]
@igor-aptos yes, already running the replay :)

File: aptos-move/aptos-vm/src/aptos_vm.rs [hunk 1]
```diff
@@ -802,18 +798,13 @@ impl AptosVM {
...
-        if self.gas_feature_version >= 15 {
-            let module_id = traversal_context
-                .referenced_module_ids
-                .alloc(entry_fn.module().clone());
-            session.check_dependencies_and_charge_gas(gas_meter, traversal_context, [(
-                module_id.address(),
-                module_id.name(),
-            )])?;
-        }
+        let module_id = traversal_context
```
Comment (vgao1996) [inline aptos-move/aptos-vm/src/aptos_vm.rs:801]: These are very recent gas changes. We can't simply remove the feature gating in this case.

Comment (georgemitenkov) [reply aptos-move/aptos-vm/src/aptos_vm.rs:801]: Offline you said we can if replay passes😄 ok, will revert

Comment (georgemitenkov) [reply aptos-move/aptos-vm/src/aptos_vm.rs]: All gas changes reverted

**Comment by georgemitenkov** [comment]
@igor-aptos I think we should be fine: https://github.com/aptos-labs/aptos-core/actions/runs/9194823097, though one job times out and rerunning doesn't help here...

**Review by gelash — APPROVED** [review]

**Comment by georgemitenkov** [comment]
cc @igor-aptos

**Review by igor-aptos — APPROVED** [review]

**Pull request merged**